### PR TITLE
Fix/old error message comes back after going to landing page

### LINKED ISSUE #282: Old error message pops up with new trajectory
## Description

When I fail to load a trajectory, go back to the landing page, then load another trajectory by clicking on a card, the previous error message shows up.

## Expected Behavior

I shouldn't see any irrelevant, old error messages popping up for a new trajectory.

## Reproduction

1. Go to staging viewer and load something that will fail, like an .obj geometry file by itself
2. Go to the landing page by clicking on "Simularium Home" in the nav bar (NOT by clicking the back button)
3. Wait for the error message in the upper right hand corner, if it's not gone already (We should probably also kill the error message once we leave the viewer page... that might be a different ticket)
3. Click on a trajectory card to load a different trajectory
4. The trajectory loads successfully, but the old error message about the .obj geometry file pops back up on the upper right hand corner!

## Details
I know why this is happening, we need to not return an `ErrorNotification` at `ViewerPanel` component mount. Returning `ErrorNotification` at component update is sufficient.

### PR DESCRIPTION
**Pull request by schoinh**
Problem
=======
Closes #282 

Solution
========
Don't render an error notification at `ViewerPanel` component mount -- it is rendered at component update, and that is enough.

## Type of change
Please delete options that are not relevant.

* Bug fix (non-breaking change which fixes an issue)

Steps to Verify:
----------------
Try to reproduce the error described in the issue -- you shouldn't be able to.

### CODE REVIEW

**Review by toloudis — APPROVED** [review]

File: src/containers/ViewerPanel/index.tsx [hunk 1]
```diff
@@ -127,7 +127,6 @@ class ViewerPanel extends React.Component<ViewerPanelProps, ViewerPanelState> {
     }
 
     public componentDidMount() {
```
Comment (meganrm) [inline src/containers/ViewerPanel/index.tsx:129]: but doesn't this mean there is still erroneously an old error in state? Even though we aren't showing it?

**Review by EricJIsaac — APPROVED** [review]

File: src/containers/ViewerPanel/index.tsx (hunk 1 above)
Comment (schoinh) [reply src/containers/ViewerPanel/index.tsx:129]: That is a good point. I was thinking that whenever a viewer loads a new trajectory it does clear out the error, but it just happens after `componentDidMount`. That is true, but it doesn't account for when you leave the viewer with errors and return to the viewer without loading a new trajectory. I'm about to push up another commit

**Comment by schoinh** [comment]
Let me know what you guys think of the most recent commit (which is meant to address @meganrm 's concern). Thanks!

File: src/index.tsx [hunk 2]
```diff
@@ -34,31 +32,12 @@ function useLocationChange() {
...
+        if (location.pathname !== VIEWER_PATHNAME) {
+            batch(() => {
+                // if we're navigating away from the viewer, stop playing and reset state
+                dispatch(setIsPlaying(false));
                 dispatch(clearSimulariumFile({ newFile: false }));
-            }
-        } else {
-            // if we're navigating away from the viewer, stop playing
-            dispatch(setIsPlaying(false));
+            });
```
Comment (schoinh) [inline src/index.tsx:40]: Essentially, I'm proposing clearing out the state when leaving the viewer page instead of when arriving at the viewer page.

**Review by EricJIsaac — APPROVED** [review]

File: src/index.tsx (hunk 2 above)
Comment (meganrm) [reply src/index.tsx:40]: this seems a lot cleaner. I had to think a bit about if this misses anything, but it seems good? Have you tested a lot of different navigation combos? Like loading a file, going back to home, loading a file from home, etc?

Comment (schoinh) [reply src/index.tsx:40]: Cool. Yeah, I haven't been able to break it yet :)

**Review by meganrm — APPROVED** [review]

**Pull request merged**